src/service.rs

```diff
@@ -580,7 +556,14 @@ where
 
     /// Spawn protocol handle
     #[inline]
-    fn session_handles_open(&mut self, id: SessionId) {
+    fn session_handles_open(
+        &mut self,
+        id: SessionId,
+    ) -> Vec<(
+        Option<futures::channel::oneshot::Sender<()>>,
```

Comment:
When this option will be none?

Reply:
> When this option will be none?

https://github.com/nervosnetwork/tentacle/blob/Remove-protocol-on-service/src/session.rs#L827 when session try to drop, wait for handles drop, it may or may not None